script.js

```diff
@@ -1,12 +1,247 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
+let allShows;
+let allEpisodes;
+
+async function setup() {
+  allShows = getAllShows();
+  const selectedShowId = allShows[0].id; // Selecting  the first show by default
+  allEpisodes = await getEpisodes(selectedShowId);
+
+  createHeader();
+  showSelectField(allShows);
+  episodeSelectField(allEpisodes);
+  addSearchField();
   makePageForEpisodes(allEpisodes);
+  createFooter();
+}
+
+// Fetching episodes for shows
+async function getEpisodes(showId) {
+  try {
+    const response = await fetch(`https://api.tvmaze.com/shows/${showId}/episodes`);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error('Error fetching episodes:', error);
+  }
+}
+
+// level 400 -adding show selector
+function showSelectField(shows) {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  const showContainer = document.createElement("div");
+  showContainer.classList.add("show-container");
+
+  const showSelect = document.createElement("select");
+  showSelect.classList.add("show-select");
+  showSelect.addEventListener("change", () => {
+    const selectedShowId = showSelect.value;
+    getEpisodes(selectedShowId)
+      .then((episodes) => {
+        allEpisodes = episodes;
+        makePageForEpisodes(allEpisodes);
+        updateEpisodeSelect(allEpisodes);
+      })
+      .catch((error) => console.error(error));
+  });
+
+  shows.forEach((show) => {
+    const option = document.createElement("option");
+    option.value = show.id;
+    option.textContent = show.name;
+    showSelect.appendChild(option);
+  });
+
+  showContainer.appendChild(showSelect);
+  searchSelectDiv.appendChild(showContainer);
+}
+
+//updating the select field to get episodes from all shows
+function updateEpisodeSelect(episodes) {
+  const episodeSelect = document.querySelector(".episode-select");
+  episodeSelect.innerHTML = "";
+
+  const selectOption = document.createElement("option");
+  selectOption.value = "select-episode";
+  selectOption.textContent = "Select Episode";
+  episodeSelect.appendChild(selectOption);
+
+  episodes.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = episode.id;
+    option.textContent = `${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    episodeSelect.appendChild(option);
+  });
+}
+
+
+
+//Creating the header of page
+function createHeader() {
+  const body = document.body;
+
+  // create the header element
+  const header = document.createElement("header");
```

Comment:
I think you are right that it makes more sense to edit the .html directly instead of adding the elements in javascript here. This is because it is harder to visualise what you are actually adding when doing it in javascript.

However, when you need to edit the DOM in real time, then you have no choice but to do it through Javascript. When we learnt React in a few weeks you'll see that adding elements is much easier in React.

Reply:
i confirmed with Shadi and i believe we have to do everything in DOM so we can get used with it